src/fides/api/models/application_config.py

```diff
@@ -184,21 +198,42 @@ def update_config_set(cls, db: Session, config: FidesConfig) -> ApplicationConfi
 
     @classmethod
     def get_resolved_config_property(
-        cls, db: Session, config_property: str, default_value: Any = None
+        cls,
+        db: Session,
+        config_property: str,
+        merge_values: bool = False,
+        default_value: Any = None,
     ) -> Optional[Any]:
         """
         Gets the 'resolved' config property based on api-set and config-set configs.
         `config_property` is a dot-separated path to the config property,
         e.g. `notifications.notification_service_type`.
 
-        Api-set values get priority over config-set, in case of conflict.
+        Api-set values get priority over config-set, in case of conflict, unless `merge_values`
+        is specified as `True`, in which case the proxy will attempt to _merge_ the api-set
+        and config-set values. Note that only iterable config values (e.g. lists) can be merged!
         """
         config_record = db.query(cls).first()
         if config_record:
             api_prop = get(config_record.api_set, config_property)
+            config_prop = get(config_record.config_set, config_property, default_value)
+
+            # if no api-set property found, fall back to config-set
             if api_prop is None:
-                logger.debug(f"No API-set {config_property} property found")
-                return get(config_record.config_set, config_property, default_value)
+                return config_prop
+
+            # if we want to merge values AND we have iterable api and config-set properties
+            # then we try to merge
+            if (
+                merge_values
+                and config_prop
+                and isinstance(api_prop, Iterable)
+                and isinstance(config_prop, Iterable)
+            ):
```

Comment:
the boolean logic here's a +little_ bit 🤔 , but it was the clearest i could come up with after a bit of time-boxed thought, and it feels OK to me. open to other suggestions!

Reply:
Hmm, there's a bit of a surprise here right? If I set `merge_values=true` but the props aren't both `Iterable` it doesn't cause an error, it just returns the `api_prop` version which might be surprising?

Is there something we can do to warn/error/etc here, or am I overthinking?

Reply:
nope i think i agree with your instinct to alert on unexpected behavior, i _think_ an error is likely appropriate 👍

Reply:
updated to throw an error in https://github.com/ethyca/fides/pull/4584/commits/23b30d2c04c982f736725419881500b8175b81c6